From e47907cc6b6eead2ab5ef4d5015445668822612a Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Mon, 13 Jul 2015 10:51:47 +0200
Subject: [PATCH] dracut-systemd/dracut-initqueue: only start service if really
 needed

Only start the dracut-initqueue.service, if the dracut_need_initqueue
command was issued on creation or later on set.
---
 modules.d/90lvm/module-setup.sh                     | 2 ++
 modules.d/90mdraid/module-setup.sh                  | 1 +
 modules.d/98dracut-systemd/dracut-initqueue.service | 5 -----
 3 files changed, 3 insertions(+), 5 deletions(-)

diff --git a/modules.d/90lvm/module-setup.sh b/modules.d/90lvm/module-setup.sh
index 8269078..42dda57 100755
--- a/modules.d/90lvm/module-setup.sh
+++ b/modules.d/90lvm/module-setup.sh
@@ -140,4 +140,6 @@ install() {
                       cache_dump cache_restore cache_check cache_repair \
                       era_check era_dump era_invalidate era_restore
     fi
+
+    dracut_need_initqueue
 }
diff --git a/modules.d/90mdraid/module-setup.sh b/modules.d/90mdraid/module-setup.sh
index 621d0fb..cecf3bf 100755
--- a/modules.d/90mdraid/module-setup.sh
+++ b/modules.d/90mdraid/module-setup.sh
@@ -132,4 +132,5 @@ install() {
         fi
     fi
     inst_hook pre-shutdown 30 "$moddir/mdmon-pre-shutdown.sh"
+    dracut_need_initqueue
 }
diff --git a/modules.d/98dracut-systemd/dracut-initqueue.service b/modules.d/98dracut-systemd/dracut-initqueue.service
index 5d772f7..fa5ca56 100644
--- a/modules.d/98dracut-systemd/dracut-initqueue.service
+++ b/modules.d/98dracut-systemd/dracut-initqueue.service
@@ -17,11 +17,6 @@ After=systemd-udev-trigger.service
 Wants=systemd-udev-trigger.service
 ConditionPathExists=/usr/lib/initrd-release
 ConditionPathExists=|/lib/dracut/need-initqueue
-ConditionPathExistsGlob=|/lib/dracut/hooks/initqueue/*.sh
-ConditionPathExistsGlob=|/lib/dracut/hooks/initqueue/settled/*.sh
-ConditionPathExistsGlob=|/lib/dracut/hooks/initqueue/finished/*.sh
-ConditionPathExistsGlob=|/lib/dracut/hooks/initqueue/online/*.sh
-ConditionPathExistsGlob=|/lib/dracut/hooks/initqueue/timeout/*.sh
 ConditionKernelCommandLine=|rd.break=initqueue
 
 [Service]
